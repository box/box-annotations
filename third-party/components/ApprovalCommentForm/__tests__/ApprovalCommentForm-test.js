/* eslint-disable no-unused-expressions */
import * as React from 'react';
import { shallow } from 'enzyme';
import noop from 'lodash/noop';

import ApprovalCommentForm from '../ApprovalCommentForm';

const fn = jest.fn();

describe('third-party/components/ApprovalCommentForm', () => {
    const render = (props = {}) =>
        shallow(
            <ApprovalCommentForm
                user={null}
                isOpen={false}
                isEditing={true}
                createComment={fn}
                onCancel={fn}
                onSubmit={noop}
                onFocus={fn}
                getAvatarUrl={noop}
                {...props}
            />
        );

    test('should correctly render empty form', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

    test('should render commentText value in textarea', () => {
        const wrapper = render().dive();

        wrapper.instance().onFormChangeHandler({ commentText: 'something' });
        wrapper.update();

        expect(wrapper.find('.bcs-comment-input').props().value).toEqual('something');
    });

    test('should return translated error message on invalid textarea message', () => {
        const intl = { formatMessage: jest.fn() };
        const wrapper = render({ intl }).dive();
        const instance = wrapper.instance();

        expect(instance.validateTextArea('peanuts')).toBeNull();
        expect(instance.validateTextArea('    ')).not.toBeNull();
    });
});
