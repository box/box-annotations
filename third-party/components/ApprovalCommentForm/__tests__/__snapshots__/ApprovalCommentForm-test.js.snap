// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`third-party/components/ApprovalCommentForm should correctly render empty form 1`] = `
<ApprovalCommentForm
  createComment={[MockFunction]}
  getAvatarUrl={[Function]}
  intl={
    Object {
      "formatDate": [Function],
      "formatMessage": [Function],
    }
  }
  isEditing={true}
  isOpen={false}
  onCancel={[MockFunction]}
  onFocus={[MockFunction]}
  onSubmit={[Function]}
  user={null}
/>
`;
